### PR TITLE
Add initial upgrade testing for Data Science Pipelines

### PR DESCRIPTION
**Pull request by jgarciao**
- Adds initial upgrade testing for Data Science Pipelines, creating a pipeline server and starting a long running pipeline, and checking the status after upgrade. 
   - Note: Aditional scenarios, like setting up an scheduled run before upgrade, will be covered in follow-up PRs
- Use `Verify Run Status` in DataSciencePipelinesBackend
- Set pip_trusted_host in gpu availability sample pipeline

### CODE REVIEW

**Review by diegolovison — APPROVED** [review]

File: ods_ci/tests/Resources/CLI/DataSciencePipelines/DataSciencePipelinesBackend.resource [hunk 1]
```diff
@@ -42,10 +43,16 @@ Create Pipeline Server
     ...    -p OBJECT_STORAGE_REGION=${object_storage_region}
     ...    -p OBJECT_STORAGE_BUCKET=${object_storage_bucket_name}
 
-    Run      oc process -f ${DSPA_PATH}/dspa-template.yaml ${template_parameters} | oc apply -n ${namespace} -f -
+    Run And Verify Command    oc process -f ${DSPA_PATH}/dspa-template.yaml ${template_parameters} | oc apply -n ${namespace} -f -    # robocop: off=line-too-long
 
     IF  ${configure_pip_index}   Create Pipelines ConfigMap With Custom Pip Index Url And Trusted Host  ${namespace}
 
+Get Dsp Version
```
Comment (FedeAlonso) [inline ods_ci/tests/Resources/CLI/DataSciencePipelines/DataSciencePipelinesBackend.resource:50]:
```suggestion
Get DSP Version
```

File: ods_ci/tests/Resources/CLI/DataSciencePipelines/DataSciencePipelinesBackend.resource [hunk 2]
```diff
@@ -106,12 +117,52 @@ Verify Pipeline Server Deployments    # robocop: disable
...
+    Verify Deployment    ${mariadb}  1  1  ${containerNames}
+
 Wait Until Pipeline Server Is Deployed
     [Documentation]    Waits until all the expected pods of the pipeline server
     ...                are running
     [Arguments]    ${namespace}
-    Wait Until Keyword Succeeds    10 times    10s
-    ...    Verify Pipeline Server Deployments    namespace=${namespace}
+
+    ${dspVersion}=    Get Dsp Version    ${namespace}
```
Comment (FedeAlonso) [inline ods_ci/tests/Resources/CLI/DataSciencePipelines/DataSciencePipelinesBackend.resource:157]:
```suggestion
    ${dspVersion}=    Get DSP Version    ${namespace}
```

File: ods_ci/tests/Resources/CLI/DataSciencePipelines/DataSciencePipelinesUpgradeTesting.resource [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    [Documentation]    Creates project dsp-test-upgrade and sets up the resources to test during upgrade:
+    ...    - Creates a pipeline server
+    ...    - Starts a pipeline that will run for 1h
+
+    Create Project And Configure Pipeline Server    ${PROJECT}
+    Start Long Running Pipeline    ${PROJECT}
+
+Verify Resources After Upgrade
+    [Documentation]    Verifies the status of the resources created in ${PROJECT} after the upgrade
+    ...    Deletes ${PROJECT} if all verifications are correct (leaving for debugging purpuses if not)
```
Comment (FedeAlonso) [inline ods_ci/tests/Resources/CLI/DataSciencePipelines/DataSciencePipelinesUpgradeTesting.resource:23]:
```suggestion
    ...    Deletes ${PROJECT} if all verifications are correct (leaving for debugging purposes if not)
```

**Review by FedeAlonso — CHANGES_REQUESTED** [review]

File: ods_ci/tests/Resources/CLI/DataSciencePipelines/DataSciencePipelinesBackend.resource (hunk 1 above)
Comment (jgarciao) [reply ods_ci/tests/Resources/CLI/DataSciencePipelines/DataSciencePipelinesBackend.resource]: Done

File: ods_ci/tests/Resources/CLI/DataSciencePipelines/DataSciencePipelinesBackend.resource (hunk 2 above)
Comment (jgarciao) [reply ods_ci/tests/Resources/CLI/DataSciencePipelines/DataSciencePipelinesBackend.resource]: Done

**Review by diegolovison — DISMISSED** [review]

**Review by FedeAlonso — DISMISSED** [review]

**Comment by jgarciao** [comment]
This PR (before the latest rebase) was tested successfully in rhoai-test-flow/617

**Review by FedeAlonso — APPROVED** [review]

**Review by diegolovison — APPROVED** [review]

**Pull request merged**